backend/gatekeeper/views.py

```diff
@@ -123,48 +135,68 @@ def index(request, path):
                         instance.name = created_object_data.get('name')
                         instance.description = created_object_data.get('description')
 
-                        if user.is_authenticated:
-                            instance.user = user
-
+                        instance.user = request.user
                         instance.save()
 
-                        if user.is_authenticated:
-                            assign_perm('subscribe_datastream', user, instance)
-                            assign_perm('publish_datastream', user, instance)
+                        assign_perm('subscribe_datastream', request.user, instance)
+                        assign_perm('publish_datastream', request.user, instance)
 
                     except Thing.DoesNotExist:
                         # TODO: handle error
                         pass
 
-    content = r.content
-    content_type = r.headers['Content-Type'] if 'Content-Type' in r.headers else ''
+    def handle_request(self, request):
+        sts_response = requests.request(request.method, self.sts_url, **self.sts_arguments)
+        status_code = sts_response.status_code
+        content_type = sts_response.headers['Content-Type'] if 'Content-Type' in sts_response.headers else ''
+
+        response = Response(content_type=content_type, status=sts_response.status_code)
+
+        if status_code == 404 or status_code >= 500:
+            response.data = sts_response.content
+            return response
+
+        if status_code == 201 and 'location' in sts_response.headers:
+            self.create(request, sts_response)
+
+        try:
+            data = sts_response.json(object_pairs_hook=OrderedDict, encoding=sts_response.encoding)
+            data = self.remap_response_content_urls(data)
+            response.data = data
+        except json.decoder.JSONDecodeError:
+            response.data = sts_response.content
+
+        headers = self.remap_response_headers(sts_response.headers)
+        if headers:
+            for name, value in headers.items():
+                response[name] = value
 
-    # If the content is JSON fix the urls to point to this view, not directly to the STS
-    # (Filter 404 out because FraunhoferIOSB/SensorThingsServer will return 404 with the
-    # content type application/json with the content "Nothing found".)
-    if cgi.parse_header(content_type)[0] == 'application/json' and r.status_code != 404:
-        data = json.loads(content, object_pairs_hook=OrderedDict, encoding=r.encoding)
+        return response
 
+    def remap_response_content_urls(self, data):
```

Comment:
These URL rewrites can be removed as we changed the URL in the STS settings

Reply:
Removed